graph/path/yen_ksp.go

```diff
@@ -70,7 +72,7 @@ func YenKShortestPaths(g graph.Graph, k int, s, t graph.Node) [][]graph.Node {
 			}
 
 			spath, weight := DijkstraFrom(spur, yk).To(t.ID())
-			if math.IsInf(weight, 1) {
+			if weight > cost {
```

Comment:
I think this should probably be 
```
			if weight > cost || math.IsInf(weight, 1) { {
```

Reply:
That would change the outcome only when weight=inf and cost=inf or nan. Both values of cost can be interpreted as "don't take cost into account", in which case the previous behavior should be (?) preserved, so adding `|| math.Inf(weight, 1)` would make sense.